.github/workflows/publish-backend-plugin-manager.yaml

```diff
@@ -50,21 +50,36 @@ jobs:
           else
             GIT_REF=master
           fi
-          echo "GIT_REF=$GIT_REF" >> $GITHUB_ENV
+          echo "GIT_REF=$GIT_REF" >> $GITHUB_OUTPUT
 
-      - name: Checkout
+      - name: Checkout from master
+        if: steps.get_checkout_ref.outputs.GIT_REF == 'master'
         uses: actions/checkout@v4
         with:
           repository: backstage/backstage
-          ref: ${{ steps.get_checkout_ref.outputs.GIT_REF }}
+          ref: master
           # fetch all history, but only for the current ref
           fetch-depth: 2147483647
           fetch-tags: false
 
+      - name: Checkout from a git ref
```

Comment:
instead of an "if master checkout from master, if other checkout from other" ... why not just ALWAYS checkout from `steps.get_checkout_ref.outputs.GIT_REF` ? Seems like on line 53 you set a value to either the `master` branch or the input branch.

Reply:
In the case of master, we also want to specify special values for `fetch-depth` and `fetch-tags`, which do not make sense for the git_ref case.